src/executor/top_n.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2022 RisingLight Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Ordering;
+
+use binary_heap_plus::BinaryHeap;
+
+use super::*;
+use crate::array::DataChunk;
+use crate::binder::BoundOrderBy;
+
+/// The executor of a Top N operation.
+pub struct TopNExecutor {
+    pub child: BoxedExecutor,
+    pub offset: usize,
+    pub limit: usize,
+    pub comparators: Vec<BoundOrderBy>,
+}
+
+impl TopNExecutor {
+    #[try_stream(boxed, ok = DataChunk, error = ExecutorError)]
+    pub async fn execute(self) {
+        let heap_size = self.offset + self.limit;
+        let mut heap = BinaryHeap::with_capacity_by(heap_size, |row1, row2| {
+            cmp(row1, row2, &self.comparators)
+        });
+
+        // collect all chunks
+        let mut chunks = vec![];
+        #[for_await]
+        for batch in self.child {
+            chunks.push(batch?);
+        }
```

Comment:
I wanted to update heap inside the loop here, but it seems impossible since the rows are references (so we cannot drop the batches).🤔

Reply:
I think it's okay to store rows with ownership inside the heap?

Reply:
Is there a type for owned rows? I only find `RowRef`

Reply:
Oh, do you mean `rows.values().collect`: `Vec<DataValue>`